packages/expo-updates/ios/EXUpdates/AppLoader/AppLoader.swift

```diff
@@ -325,13 +323,24 @@ open class AppLoader: NSObject {
           // the database and filesystem have gotten out of sync
           // do our best to create a new entry for this file even though it already existed on disk
           // TODO: we should probably get rid of this assumption that if an asset exists on disk with the same filename, it's the same asset
-
-          // this has always force-tried implicitly, could probably do some better error handling
-          // swiftlint:disable:next force_try
-          let contents = try! Data(contentsOf: self.directory.appendingPathComponent(existingAsset.filename))
-          existingAsset.contentHash = UpdatesUtils.hexEncodedSHA256WithData(contents)
-          existingAsset.downloadTime = Date()
-          self.finishedAssets.append(existingAsset)
+          var contents: Data?
+          do {
+            contents = try Data(contentsOf: self.directory.appendingPathComponent(existingAsset.filename))
+          } catch {}
+          if contents == nil {
+            do {
+              if let embeddedUrl = UpdatesUtils.url(forBundledAsset: existingAsset) {
+                contents = try Data(contentsOf: embeddedUrl)
+              }
+            } catch {}
+          }
```

Comment:
```suggestion
          let contents = try Data(contentsOf: self.directory.appendingPathComponent(existingAsset.filename)) ??
            UpdatesUtils.url(forBundledAsset: existingAsset).let { it in try? Data(contentsOf: it) }
```

Reply:
The above change will break OTA updates, as this code runs for both OTA updates and embedded bundles. The code checks the updates folder, and if the asset is not found there, checks application resources.

Reply:
Ah, sorry, it should be `try?` on the first one. edited.

Reply:
The main thing is that a naked do/catch is equivalent to `try?`

Reply:
Ah thanks for the explanation!

Reply:
Same comment (looks like the change wasn't applied?)

Reply:
When I applied it I started getting crashes. I just rebased on top of your refactor PRs, once CI runs I'll test putting this change back in.

Reply:
Made this change -- the logic is not quite correct and does crash. I made a change to keep the `try?` but fix the logic, and updates flows (including excluded assets flow) now work as expected.